howso/react_aggregate.amlg

```diff
@@ -372,360 +380,299 @@
-								)
-						))
-					)
-				)
-
-				(assign_to_entities (assoc
-					!residualsMap residuals_map
-					!featurePredictionStatsMap prediction_stats_map
-				))
-				(accum_to_entities (assoc !revision 1))
-			)
-		)
-
-		;set the custom hp map if specified a hyperparameter_param_path
 		(declare (assoc
+			;set the custom hp map if specified a hyperparameter_param_path
 			custom_hyperparam_map
 				(if hyperparameter_param_path
 					(get !hyperparameterMetadataMap hyperparameter_param_path)
 				)
+			;initialize conditional queries
+			action_condition_filter_query
+				(if (get details "action_condition")
+					(call !GetQueryByCondition (assoc
+						condition (get details "action_condition")
+						precision (or (get details "action_condition_precision") "exact")
+						num_cases (get details "action_num_samples")
+					))
+					(list)
+				)
+			context_condition_filter_query
+				(if (get details "context_condition")
+					(call !GetQueryByCondition (assoc
+						condition (get details "context_condition")
+						precision (or (get details "context_condition_precision") "exact")
+						num_cases (get details "context_num_samples")
+					))
+					(list)
+				)
 		))
 
-		;condtioned_residuals is set to true if any form of condition stats are provided and any prediction stats are selected
-		(if (and
-				(or (get details "context_condition") (get details "action_condition") (get details "prediction_stats_features"))
-				(or (get details "prediction_stats") (get details "prediction_stat_robust"))
+		;calculates and stores the full residuals unless its conditioned prediction stats
+		(if (get details "prediction_stats")
+			(let
+				(assoc
+					prediction_stats_map
+						(call !CalculateFeatureResiduals (assoc
+							features action_features
+							context_features context_features
+							robust_residuals (false)
+							hyperparameter_feature prediction_stats_action_feature
+							use_case_weights use_case_weights
+							weight_feature weight_feature
+							custom_hyperparam_map custom_hyperparam_map
+							compute_all_statistics (true)
+							confusion_matrix_min_count confusion_matrix_min_count
+							action_condition_filter_query action_condition_filter_query
+							context_condition_filter_query context_condition_filter_query
+							case_ids
+								;if there is an action condition, case ids matching the condition will be selected internally
+								(if (size action_condition_filter_query)
+									(null)
+									case_ids
+								)
+						))
+					selected_prediction_stats
+						(call !ProcessSelectedPredictionStats (assoc selected_prediction_stats (or (get details "selected_prediction_stats") [])))
+				)
+				(accum (assoc
+					output
+						(append
+							(filter
+								(lambda (contains_value selected_prediction_stats (current_index)))
+								(remove (get prediction_stats_map "prediction_stats") [".robust" ".hyperparam_path"])
```

Comment:
can this ever result in a null? cos (append (null) {} {} ) will result in a badly formatted assoc

Reply:
Pretty sure it cannot result in a null, `(get prediction_stats_map "prediction_stats")` will always return an (assoc) when passing `compute_all_statistics (true)`